packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -100,17 +105,86 @@ abstract class SelectionHandler implements ValueListenable<SelectionGeometry> {
   SelectionResult dispatchSelectionEvent(SelectionEvent event);
 }
 
+/// This class stores the information of an active selection under
+/// a [Selectable] or [SelectionHandler], including the start and
+/// end positions of a selection relative to the content, as well
+/// as an identifier to identify the owner of the content, any
+/// children selection ranges that may be contained within the active
+/// selection, and the content itself.
+///
+/// [SelectionArea] and [SelectableRegion] provide access to this
+/// information through the [SelectionListener] passed to their
+/// [SelectionListener.onSelectionChanged] member.
+///
+/// See also:
+///
+///   * [SelectionListener], which provides the [SelectedContentRange]s representing
+///     the active selection for a given subtree contained under a [SelectionArea] 
+///     or [SelectableRegion].
+abstract class SelectedContentRange<T extends Object> {
+  /// Creates a range for the content of a [Selectable] or [SelectionHandler].
+  SelectedContentRange({
+    this.selectableId,
+    required this.content,
+    this.children,
+  });
+
+  /// The unique id for the [Selectable] that created the range.
+  final Object? selectableId;
+
+  /// The content that contains the selection.
+  final T content;
+
+  /// The start of the selection relative to the [content].
+  int get startOffset;
+
+  /// The end of the selection relative to the [content].
```

Comment:
I feel like somewhere you need a real world example of what these values would be. Maybe in the main SelectedContentRange do something like: Consider the string "hello world" where "world" is bold and selected. This could be represented by the [content] of a [TextSpan] where [TextSpan.text] is "hello " and [TextSpan.children] consists of a single [TextSpan] with [TextSpan.text] "world" and bold styling. In that case, the [startOffset] would be 6 and [endOffset] would be 11.

And also explain what SelectedContentRange.children would be.

Or maybe show the values in a code block instead of giving them in paragraph form like I did...

Reply:
Added an example in the docs.

Reply:
Thank you that example helps a lot :+1: